src/main/java/com/uid2/shared/health/HealthManager.java

```diff
@@ -1,14 +1,27 @@
 package com.uid2.shared.health;
 
-import java.io.File;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 
 public class HealthManager {
     public static HealthManager instance = new HealthManager();
-    private AtomicReference<List<IHealthComponent>> componentList = new AtomicReference(new ArrayList<IHealthComponent>());
+    private final AtomicReference<List<IHealthComponent>> components = new AtomicReference<>(new ArrayList<>());
+    private MeterRegistry registry = null;
+
+    public synchronized void setMeterRegistry(MeterRegistry newRegistry) {
```

Comment:
I think we can just use Metrics.globalRegistry?

Reply:
Changed to global.